File: 1000-9999/1019.py
s=input()
cnt=[0 for _ in range(10)]

if int(s)<10:
    for i in range(1, int(s)+1):
        cnt[i]+=1
    print(*cnt)
elif int(s)==10:
    print("1 2 1 1 1 1 1 1 1 1")
else:
    for i in range(len(s)):
        for j in range(10):
            if i==0 and j==0: continue
            if int(s[i])!=j:
                if i==0: 
                    if int(s[i])>j:
                        cnt[j]+=pow(10, len(s)-1)
                else:
                    tmp=""
                    for k in range(i): tmp+=s[k]
                    if int(s[i])>j and j!=0:
                        tmp=int(tmp)+1
                    tmp=int(tmp)
                    for k in range(len(s)-i-1): tmp*=10
                    cnt[j]+=tmp
            elif int(s[i])==j:
                if int(s[i])==0:
                    tmp=""
                    for k in range(len(s)):
                        if k==i: 
                            tmp=str(int(tmp)-1)
                            continue
                        tmp+=s[k]
                    cnt[0]+=int(tmp)+1
                else:
                    tmp=""
                    for k in range(len(s)):
                        if k==i: continue
                        tmp+=s[k]
                    tmp=int(tmp)
                    if j!=0: tmp+=1
                    cnt[j]+=tmp

    print(*cnt)

File: 1000-9999/1019.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long int

string s;
ll ans[10];

void solve(int cur){
    if(cur==s.size()) return;

    for(int i=0; i<10; i++){
        if(cur==0 && i==0) continue;
        if((s[cur]-'0')>i){
            ll p=1, tmp=0, pp=pow(10, s.size()-1);
            if(cur==0) ans[i]+=pp;
            else{
                for(int j=cur-1; j>=0; j--){
                    tmp+=p*(s[j]-'0');
                    p*=10;
                } tmp+=(i==0? 0:1); 
                while(tmp!=0 && tmp<pp) tmp*=10;
                ans[i]+=tmp/10;
            }
        } else if((s[cur]='0')==i){
            ll p=1, tmp=0;
            for(int j=s.size()-1; j>=0; j--){
                if(j==cur) continue;
                tmp+=p*(s[j]-'0');
                p*=10;
            } ans[i]+=tmp+1;
        } else{
            ll p=1, tmp=0, pp=pow(10, s.size()-1);
            for(int j=cur-1; j>=0; j--){
                tmp+=p*(s[j]-'0');
                p*=10;
            } tmp+=(i==0? 0:1); 
            while(tmp!=0 && tmp<pp) tmp*=10;
            ans[i]+=tmp/10;
        }
    } solve(cur+1);
}

int main(void){
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);

    cin>>s;
    solve(0);

    ans[0]++;
    for(int i=0; i<10; i++) cout<<ans[i]<<" ";
    cout<<"\n";
}